Validated form fields

MaddieLvrg opened the pull request:
The submit buttons are disabled until all forms are completed. Will show
an error message if something is incorrect.

MaddieLvrg committed:
Validated form fields

The submit buttons are disabled until all forms are completed. Will show
an error message if something is incorrect.

kjrcda commented:
The spans is the wrong way to do this, on the main page there is already an error handler, just need to hook into that. Same idea for the register page. This also leads to not disabling the buttons, when a user clocks the button and has invalid input then they should be notified.

kjrcda committed:
api now returns incorrect password

changed api function, now it returns incorrect password error.

kjrcda committed:
Removed div's, checks for email

Now we check for email validation in the controller and simplify the
jade file. Also uses errorMessages that was already there.

kjrcda committed:
Cleaning up register page

- cleaned up spans, now use errorMessages
- new css for register boxes, should be more dynamic with resizes
- also with this commit all emails will be transmitted and saved as
lowercase
-*note* 2 of the css will not work until dev is pulled into here

kjrcda committed:
Unused name

^

kjrcda committed:
CSS improvements

- main page no longer has massive form, its limited now
- register button has max width now

kjrcda committed:
Error now has same restrictions

added same restrictions to error messages on main page as the form

paymog commented on webapp/public/javascripts/controllers.js:
```diff
@@ -26,7 +26,13 @@ lunchrControllers.controller('MainPageController', ['$scope', '$http', '$state',
-            $http.post('/api/users/authenticate', {email: $scope.email, password: $scope.password})
+            if(/^[_a-z0-9]+(\.[_a-z0-9]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/.test($scope.email) == false)
+            {
+                $scope.errorMessages = "Incorrect Email";
```
Incorrect or invalid?

paymog commented on webapp/routes/api.js:
```diff
@@ -55,17 +55,34 @@ router.post('/users/register', function (req, res, next) {
 
 router.post('/users/authenticate', function (req, res, next) {
-    User.findOne({email: req.body.email, password: req.body.password}, function (error, user) {
+    User.find({email: req.body.email}, function (error, users) {
```
Why was this changed? There should only ever be a single person with an email, right?

kjrcda replied:
This was changed so that we can tell if there is a valid email but incorrect password entered

paymog replied:
but, still do findOne and just compare passwords?

paymog replied:
like, semantically, `find` should only ever return an array of length 1 or 0

paymog commented on webapp/routes/api.js:
```diff
@@ -55,17 +55,34 @@ router.post('/users/register', function (req, res, next) {
+            }
+        }
+
+        if(found == false) {
```
use `if(!found)`

paymog commented on webapp/public/javascripts/controllers.js:
```diff
@@ -66,8 +72,14 @@ lunchrControllers.controller('RegisterController', ['$scope', '$http', '$state',
+
             $http.post('/api/users/register', {
-                email: $scope.email,
+                email: $scope.email.toLowerCase(),
```
We should lowercase on the serverside. No here. That way the ios client works with this concept too.

kjrcda committed:
Lowercase on server, password compare refined

- Invalid, not incorrect
- changed find of user to findOne (as it should be)
- toLowerCase now on server

paymog commented on webapp/routes/api.js:
```diff
@@ -63,9 +63,14 @@ router.post('/users/authenticate', function (req, res, next) {
+        }
+
         user.password = null;
         res.json(user);
```
do we have to undo this null set here?

kjrcda replied:
I don't think so. The behavior is ok in the api but wasn't in socket. I will add the resetting of the password just in case though.

paymog commented on webapp/routes/api.js:
```diff
@@ -36,7 +36,7 @@ router.post('/users/register', function (req, res, next) {
         // create and save the user
         var user = new User({
-            email: req.body.email,
+            email: req.body.email.toLowerCase(),
```
we're not lowercasing in the find statement above

kjrcda committed:
Fixed a toLowerCase and password reset
